assets/js/src/frontend.js

```diff
@@ -8,7 +12,14 @@ const intersectionObserver = new IntersectionObserver( ( entries ) => {
 			.querySelector( '.wp-block-post-template' ),
 		$clickedButton = entries[ 0 ].target;
 
-	fetchPosts( $url, $container, $clickedButton );
+	// Get the page id from url arg page=n
+	const page = $url.match( /page=(\d+)/ );
```

Comment:
Just for the sake of not using Regex and maybe also improving readability what if we tried to get the page param with something like:

```suggestion
	const page = new URL( $url ).searchParams.get( 'page' )
```

Reply:
This won't work as it's not a true url arg

`https://some.tld/post/?query-100-page=4`

However this get me to thinking i am not handling multiple queries per page, so i have made some changes to keep track of both the query id and the page number.

I have made changes to the PR and tested it the best (the plugin isnt the best with multiple queries per page)